core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -279,53 +305,69 @@ public DiscoveryNode resolveNode(String node) {
                 } else if (nodeExists(nodeId)) {
                     resolvedNodesIds.add(nodeId);
                 } else {
-                    // not a node id, try and search by name
+                    boolean matchedEphemeralId = false;
                     for (DiscoveryNode node : this) {
-                        if (Regex.simpleMatch(nodeId, node.getName())) {
+                        if (node.getEphemeralId().equals(nodeId)) {
```

Comment:
why did add this extra resolveNode path? isn't ephemeralId supposed to be internal and shouldn't be used in an API?